app/src/ui/add-repository/create-repository.tsx

```diff
@@ -566,6 +569,32 @@ export class CreateRepository extends React.Component<
     )
   }
 
+  private renderGitRepositorySubFolderMessage() {
+    const { isSubFolderOfRepository, path, name } = this.state
+
+    if (!path || path.length === 0 || !isSubFolderOfRepository) {
+      return null
+    }
+
+    const fullPath = Path.join(path, sanitizedRepositoryName(name))
+
+    return (
+      <Row>
+        <InputWarning
+          id="path-is-subfolder-of-repository"
+          trackedUserInput={this.state.path + this.state.name}
+          ariaLiveMessage={`The directory ${fullPath} appears to be a subfolder Git repository. Did you know about submodules?`}
+        >
+          The directory <Ref>{fullPath}</Ref>appears to be a subfolder of Git
+          repository.
+          <LinkButton uri="https://git-scm.com/book/en/v2/Git-Tools-Submodules">
```

Comment:
Should we use a shortener to have control of what URL it points to? (In case we change it in the future to our own docs, for example)